tests/crc32/main.c

```diff
@@ -20,6 +24,71 @@ static struct testcase {
 	    {0x1b, 0xc4, 0xb4, 0x28}}
 };
 
+/* Largest buffer must be first. */
+static const size_t perfsizes[] = {16384, 8192, 4096, 2048, 1024, 512, 256,
+    128, 64, 32, 16};
+static const size_t num_perf = sizeof(perfsizes) / sizeof(perfsizes[0]);
+static const size_t bytes_to_hash = 1 << 27;	/* approx 100 Mb */
+
+static int
+performance_test()
+{
+	CRC32C_CTX ctx;
+	uint8_t * largebuf;
+	uint8_t cbuf[4];
+	struct timeval begin, end;
+	long long delta;
+	size_t i, j;
+	size_t num_hashes;
+
+	/* Allocate buffer to hold largest message. */
+	if ((largebuf = malloc(perfsizes[0])) == NULL) {
+		warnp("malloc");
+		goto err0;
+	}
+	memset(largebuf, 0, perfsizes[0]);
+
+	/* Get beginning time. */
```

Comment:
Get the start and end time for each size.  We want to say how fast each size of CRC32 takes.